dev(lint): add :path-invalid-construct/string-join linter

RCmerci opened the pull request:
Lint those exprs that use `string/join` to construct the path:
`(string/join "/" [...])`

@andelf is working on a unified way(for different platforms) to do path-related operations.

logseq-cldwalker commented:
@RCmerci Excited for this. Labeling as on hold until it's ready for review

RCmerci commented:
@logseq-cldwalker, hi,  I updated this linter's level to :info, to make it not break the lint CI.
> @andelf is working on a unified way(for different platforms) to do path-related operations.

and this work still have a long way to go I think.
we can still be reminded in editors like this.
<img width="429" alt="image" src="https://user-images.githubusercontent.com/5608710/221353658-862d1cb9-dc27-444e-9aad-beb24437109b.png">

logseq-cldwalker reviewed (approved):
@RCmerci Nice use of analyze-call! 👍 🚢  With all the invalid string/join's, clj-kondo output is [more noisy](https://github.com/logseq/logseq/actions/runs/4269678703/jobs/7433008729) and that may make it hard to see errors. Optional feedback: It could be nice to move all these `string/join` uses to one fn to be improved later

RCmerci commented:
> @RCmerci Nice use of analyze-call! 👍 🚢 With all the invalid string/join's, clj-kondo output is [more noisy](https://github.com/logseq/logseq/actions/runs/4269678703/jobs/7433008729) and that may make it hard to see errors. Optional feedback: It could be nice to move all these `string/join` uses to one fn to be improved later

updated! added `util/string-join-path` to reduce lint output

logseq-cldwalker commented:
Thanks! 🚢
